src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -41,10 +41,214 @@ pub type caddr_t = *mut ::c_char;
 
 pub type fhandle_t = fhandle;
 
+// This is mosty a hack to keep the length somewhat similar. However, based on
+// https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html, it might very well be incorrect.
+cfg_if! {
+    if #[cfg(any(target_pointer_width = "16", target_pointer_width = "32"))] {
+        pub type c_longdouble = [u8; 8];
+    } else {
+        pub type c_longdouble = [u8; 16];
+    }
+}
```

Comment:
Have you checked that this actually works? I imagine that from a calling-convention point of view, a `long double` is passed/returned differently from an array of `u8`.

Reply:
Do you know how to check it? I tried it on the C API and valgrind seemed fine with it but no clue if it isn't just plain luck at this point...

Reply:
I did a few tests on godbolt:
- On x86 & x86_64, long doubles are passed in 80-bit x87 float registers. There is no way to represent this with normal Rust code.
- On aarch64, long doubles are passed in 128-bit SIMD registers. You can probably emulate this by using `uint64x2_t` from the (currently unstable) stdarch intrinsics.

In conclusion, I don't think it is practical to bind these API using Rust at the moment. I would recommend using a wrapper written in C to convert these values to a normal `double`.

In the longer term, I plan on adding arch-specific types such as `f80`, `f128`, `f16`, etc to stdarch, but this is unlikely to happen any time soon.

Reply:
Would it make sense to make it private and remove the double functions using `long double` and keeping it for the `statinfo` struct?

Reply:
We only test FreeBSD on one arch? :o

Reply:
We only test it on x86_64. But we do test that it builds on x86_64, i686, aarch64 and powerpc64.

Reply:
I'm not sure what you mean: are the types actually checked on the arch you mentioned or is it simply running a `cargo build`?

Reply:
The types are only checked on x86_64 I think.

Reply:
Oh, that's pretty bad. :-/